### DISCUSSION TITLE
Understanding what Position.character represents and how to convert this to Col

### DISCUSSION THREAD

**Discussion by callmephilip** [microsoft/vscode-discussions#1616] [Extension Development QnA] [answered]
I am having hard time figuring out what `Position.character` actually represents and how to convert this value into a more traditional `Col` value (i.e. column position in your editor to go along with Line). Given the following context:

```ts
const editor = vscode.window.activeTextEditor;
const charOffset = editor.selection.anchor.character;
```

How does one go from `charOffset` to column in this case? Or is there a better API for getting col value directly?

**Comment by alefragnani** [comment]
Hi @callmephilip ,

If you need to detect the column position _of the cursor_, you should use `editor.selection.active.character` instead of `anchor`, because `anchor` refers to the position _at which the selection starts_. By the docs (https://code.visualstudio.com/api/references/vscode-api#Selection)

And by the way,  both `line` and `character` attributes are _zero based_ .

Hope this helps

**Comment by callmephilip** [comment, marked as answer]
@alefragnani thanks for the suggestions. I think I have finally figured it out. Reposted my solution in the relevant SO thread https://stackoverflow.com/questions/64791832/vscode-how-can-i-determine-the-cursors-column-position-within-an-editor/78989763#78989763
